src/Surface_TEST.cc

```diff
@@ -126,3 +190,242 @@ TEST(DOMcontact, CollideBitmask)
+  EXPECT_DOUBLE_EQ(friction2.ODE()->Slip2(), 4);
+  EXPECT_EQ(friction2.ODE()->Fdir1(),
+            ignition::math::Vector3d(1, 2, 3));
+}
+
+/////////////////////////////////////////////////
+TEST(DOMfriction, CopyAssignmentOperator)
+{
+  sdf::ODE ode1;
+  ode1.SetMu(0.1);
+  ode1.SetMu2(0.2);
+  ode1.SetSlip1(3);
+  ode1.SetSlip2(4);
+  ode1.SetFdir1(ignition::math::Vector3d(1, 2, 3));
+  sdf::Friction friction1;
+  friction1.SetODE(ode1);
+
+  sdf::Friction friction2 = friction1;
+  EXPECT_DOUBLE_EQ(friction2.ODE()->Mu(), 0.1);
+  EXPECT_DOUBLE_EQ(friction2.ODE()->Mu2(), 0.2);
+  EXPECT_DOUBLE_EQ(friction2.ODE()->Slip1(), 3);
+  EXPECT_DOUBLE_EQ(friction2.ODE()->Slip2(), 4);
+  EXPECT_EQ(friction2.ODE()->Fdir1(),
+            ignition::math::Vector3d(1, 2, 3));
+}
+
+/////////////////////////////////////////////////
+TEST(DOMfriction, CopyAssignmentAfterMove)
+{
+  sdf::ODE ode1;
+  ode1.SetMu(0.1);
+  ode1.SetMu2(0.2);
+  ode1.SetSlip1(3);
+  ode1.SetSlip2(4);
+  ode1.SetFdir1(ignition::math::Vector3d(1, 2, 3));
+  sdf::Friction friction1;
+  friction1.SetODE(ode1);
+
+  sdf::ODE ode2;
+  ode2.SetMu(0.2);
+  ode2.SetMu2(0.1);
+  ode2.SetSlip1(7);
+  ode2.SetSlip2(8);
+  ode2.SetFdir1(ignition::math::Vector3d(3, 2, 1));
+  sdf::Friction friction2;
+  friction2.SetODE(ode2);
+
+  sdf::Friction tmp = std::move(friction1);
+  friction1 = friction2;
+  friction2 = tmp;
+
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Mu(), 0.2);
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Mu2(), 0.1);
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Slip1(), 7);
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Slip2(), 8);
+  EXPECT_EQ(friction1.ODE()->Fdir1(),
+            ignition::math::Vector3d(3, 2, 1));
+  EXPECT_DOUBLE_EQ(friction2.ODE()->Mu(), 0.1);
+  EXPECT_DOUBLE_EQ(friction2.ODE()->Mu2(), 0.2);
+  EXPECT_DOUBLE_EQ(friction2.ODE()->Slip1(), 3);
+  EXPECT_DOUBLE_EQ(friction2.ODE()->Slip2(), 4);
+  EXPECT_EQ(friction2.ODE()->Fdir1(),
+            ignition::math::Vector3d(1, 2, 3));
+}
+
+/////////////////////////////////////////////////
+TEST(DOMfriction, Set)
+{
+  sdf::ODE ode1;
+  sdf::Friction friction1;
+
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Mu(), 1.0);
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Mu2(), 1.0);
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Slip1(), 0);
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Slip2(), 0);
+  EXPECT_EQ(friction1.ODE()->Fdir1(),
+            ignition::math::Vector3d(0, 0, 0));
+
+  ode1.SetMu(0.1);
+  ode1.SetMu2(0.2);
+  ode1.SetSlip1(3);
+  ode1.SetSlip2(4);
+  ode1.SetFdir1(ignition::math::Vector3d(1, 2, 3));
+  friction1.SetODE(ode1);
+
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Mu(), 0.1);
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Mu2(), 0.2);
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Slip1(), 3);
+  EXPECT_DOUBLE_EQ(friction1.ODE()->Slip2(), 4);
+  EXPECT_EQ(friction1.ODE()->Fdir1(),
+            ignition::math::Vector3d(1, 2, 3));
+}
+
+/////////////////////////////////////////////////
+TEST(DOMode, DefaultConstruction)
+{
+  sdf::ODE ode;
```

Comment:
Since the test is named `DefaultConstruction`, I think it should test the default values, not ones set using the `Set*` functions.